src/Hedgehog/Property.fs

```diff
@@ -81,11 +81,28 @@ module Property =
     let bind (k : 'a -> Property<'b>) (m : Property<'a>) : Property<'b> =
         bindGen (toGen << k) (toGen m) |> ofGen
 
+    let private printValue (value) : string =
+        // sprintf "%A" is not prepared for printing ResizeArray<_> (C# List<T>) so we prepare the value instead
+        let prepareForPrinting (value: obj) : obj =
+        #if FABLE_COMPILER
+            value
+        #else
+            let t = value.GetType()
+            // have to use TypeInfo due to targeting netstandard 1.6
+            let t = System.Reflection.IntrospectionExtensions.GetTypeInfo(t)
+            let isList = t.IsGenericType && t.GetGenericTypeDefinition() = typedefof<ResizeArray<_>>
+            if isList
+            then value :?> System.Collections.IEnumerable |> Seq.cast<obj> |> List.ofSeq :> obj
+            else value
+        #endif
+
+        value |> prepareForPrinting |> sprintf "%A"
```

Comment:
I think it might be possible to extend the `Gen<'a>` type to include a format function for the type. For most values, this could be `fun value -> sprintf "%A" value`, and for `ResizeArray<'a>` it could be `fun resizeArray -> <your list logic above>`. This would eliminate the need for reflection, and keep us from having to add special cases for each type.

@TysonMN What are your thoughts on the approach I outlined above?

Reply:
Seems reasonable.  However, I would create a new issue for that alternative design and merge this PR as is since it fixes a bug without changing the public API.

Reply:
Done, the new issues is #326.